### PR TITLE
bring in eslint and fix issues

### LINKED ISSUE #1: Linter
Add linter for css, JS
cargo clippy for rust

### PR DESCRIPTION
**Pull request by ericawright**
Fixes #1 

todo: add linter for css + rust

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -4,16 +4,23 @@
...
   },
   "scripts": {
     "start": "react-scripts start",
+    "prebuild": "npm run lint",
     "build": "react-scripts build",
     "watch": "watch 'npm run build' src public",
     "test": "react-scripts test --env=jsdom",
-    "eject": "react-scripts eject"
+    "eject": "react-scripts eject",
+    "lint": "./node_modules/.bin/eslint src/*.js"
```
Comment (bwinton) [inline client/package.json]: I think you can just type `eslint src/*.js` here. Or better, add a `.eslintignore` file containing `node_modules` and `build`, and change it to just `eslint`. 🙂

File: client/package.json [hunk 2]
```diff
@@ -4,16 +4,23 @@
   "private": true,
   "dependencies": {
     "photon-colors": "3.3.1",
+    "prop-types": "15.6.2",
     "react": "16.5.1",
     "react-dom": "16.5.1",
     "react-scripts": "1.1.5",
     "watch": "1.0.2"
```
Comment (bwinton) [inline client/package.json]: Should `watch` also be a devDependency, now that we have that section?

File: client/src/App.js [hunk 3]
```diff
@@ -1,30 +1,28 @@
-import React, { Component } from 'react';
 import './App.css';
-import Project from './Project';
+import React, {Component} from 'react';
```
Comment (bwinton) [inline client/src/App.js:2]: I would kind of have preferred this to be a separate commit, but 🤷‍♂️.

File: client/src/App.js [hunk 4]
```diff
@@ -1,30 +1,28 @@
...
-      projects: [],
-    };
+    this.state = {projects: []};
     this.submitNewProject = this.submitNewProject.bind(this);
     this.showProjects = this.showProjects.bind(this);
     this.toggleForm = this.toggleForm.bind(this);
   }
 
-  componentWillMount() {
+  UNSAFE_componentWillMount() {
```
Comment (bwinton) [inline client/src/App.js:15]: Whoa. Does this still get called if it's got a different name?

File: client/src/App.js [hunk 5]
```diff
@@ -71,25 +69,21 @@ class App extends Component {
...
   render() {
     return (
       <div className="App">
-        <Header/>
+        <Header />
         <div className="grid-wrapper">
           {/* Column 1 */}
           <h2 className="column-title" id="submitted">Submitted</h2>
-          <button onClick="" id="suggest-button" onClick={this.toggleForm}>Suggest a new project</button>
+          <button id="suggest-button" onClick={this.toggleForm}>Suggest a new project</button>
```
Comment (bwinton) [inline client/src/App.js:84]: I've fixed this at least twice in my webpack debug build experiments. 🤣

File: client/src/registerServiceWorker.js [hunk 6]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable */
```
Comment (bwinton) [inline client/src/registerServiceWorker.js:1]: Should we ever re-enable this?

**Review by bwinton — APPROVED** [review]
Mostly looks good. Some small things I think could be cleaned up before we ship it, though. 🙂

File: client/src/App.js (hunk 4 above)
Comment (ericawright) [reply client/src/App.js:15]: it's the same function, they deprecated the other - trying to make us not use it

File: client/src/App.js (hunk 3 above)
Comment (ericawright) [reply client/src/App.js:2]: this was from the linter, it wants imports to be in a certain order - not sure what about it you'd like to be separate?

File: client/src/registerServiceWorker.js (hunk 6 above)
Comment (ericawright) [reply client/src/registerServiceWorker.js:1]: I haven't decided what to do with this file yet, might remove it altogether
